static/app/components/segmentedControl.tsx

```diff
@@ -119,36 +133,46 @@ function Segment({
 
   const {isDisabled} = props;
   return (
-    <SegmentWrap size={size} isSelected={isSelected} isDisabled={isDisabled}>
-      <SegmentInput {...inputProps} ref={ref} />
-      {!isDisabled && (
-        <SegmentInteractionStateLayer
-          nextOptionIsSelected={nextOptionIsSelected}
-          prevOptionIsSelected={prevOptionIsSelected}
-        />
-      )}
-      {isSelected && (
-        <SegmentSelectionIndicator
-          layoutId={layoutGroupId}
-          transition={{type: 'tween', ease: 'easeOut', duration: 0.2}}
-          priority={priority}
-          aria-hidden
-        />
-      )}
-
-      <Divider visible={showDivider} role="separator" aria-hidden />
-
-      {/* Once an item is selected, it gets a heavier font weight and becomes slightly
-      wider. To prevent layout shifts, we need a hidden container (HiddenLabel) that will
-      always have normal weight to take up constant space; and a visible, absolutely
-      positioned container (VisibleLabel) that doesn't affect the layout. */}
-      <LabelWrap>
-        <HiddenLabel aria-hidden>{props.children}</HiddenLabel>
-        <VisibleLabel isSelected={isSelected} isDisabled={isDisabled} priority={priority}>
-          {props.children}
-        </VisibleLabel>
-      </LabelWrap>
-    </SegmentWrap>
+    <Tooltip
```

Comment:
Does it always render? can we conditionally do this

```
const stuff = (
      <SegmentWrap size={size} isSelected={isSelected} isDisabled={isDisabled}>
        <SegmentInput {...inputProps} ref={ref} />
        {!isDisabled && (
          <SegmentInteractionStateLayer
            nextOptionIsSelected={nextOptionIsSelected}
            prevOptionIsSelected={prevOptionIsSelected}
          />
        )}
        {isSelected && (
          <SegmentSelectionIndicator
            layoutId={layoutGroupId}
            transition={{type: 'tween', ease: 'easeOut', duration: 0.2}}
            priority={priority}
            aria-hidden
          />
        )}

        <Divider visible={showDivider} role="separator" aria-hidden />

        {/* Once an item is selected, it gets a heavier font weight and becomes slightly
        wider. To prevent layout shifts, we need a hidden container (HiddenLabel) that
        will always have normal weight to take up constant space; and a visible,
        absolutely positioned container (VisibleLabel) that doesn't affect the layout. */}
        <LabelWrap>
          <HiddenLabel aria-hidden>{props.children}</HiddenLabel>
          <VisibleLabel
            isSelected={isSelected}
            isDisabled={isDisabled}
            priority={priority}
          >
            {props.children}
          </VisibleLabel>
        </LabelWrap>
      </SegmentWrap>
)

if (tooltip) {
  return <Tooltip ...>{stuff}</Tooltip>
}

return stuff
```

Otherwise in the dev tools we just have to constantly see tooltip components in the react tree.